mirrord/kube/src/api/kubernetes.rs

```diff
@@ -227,18 +228,31 @@ impl KubernetesAPI {
             progress.warning(format!("Target has multiple containers, mirrord picked \"{container_name}\". To target a different one, include it in the target path.").as_str());
         }
 
-        let incoming_mode = config.map(|config| config.feature.network.incoming.mode);
-        let is_mesh = runtime_data
-            .as_ref()
-            .map(|data| data.mesh.is_some())
-            .unwrap_or_default();
-        if matches!(incoming_mode, Some(IncomingMode::Mirror)) && is_mesh {
-            progress.warning(
-                "mirrord has detected that you might be running on a cluster with a \
-                 service mesh and `network.incoming.mode = \"mirror\"`, which is currently \
-                 unsupported. You can set `network.incoming.mode` to \"steal\" (check out the\
-                 `http_filter` configuration value if you only want to steal some of the traffic).",
-            );
+        if let Some(mesh) = runtime_data.as_ref().and_then(|data| data.mesh.as_ref()) {
+            progress.info(&format!("service mesh detected: {mesh}"));
+
+            let incoming_mode = config.map(|config| config.feature.network.incoming.mode);
+
+            if matches!(incoming_mode, Some(IncomingMode::Mirror)) {
+                progress.warning(
+                    "mirrord has detected that you might be running on a cluster with a \
+                     service mesh and `network.incoming.mode = \"mirror\"`, which is currently \
+                     unsupported. You can set `network.incoming.mode` to \"steal\" (check out the\
+                     `http_filter` configuration value if you only want to steal some of the traffic).",
```

Comment:
Huh, wait. Is this still true? :eyes: 
We even have custom logic in the sniffer when mesh is detected

Reply:
Yep it works now, I removed it